bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/serialization/BoschServiceDataDeserializer.java

```diff
@@ -42,35 +43,27 @@ public class BoschServiceDataDeserializer implements JsonDeserializer<BoschSHCSe
     public BoschSHCServiceState deserialize(JsonElement jsonElement, Type type,
             JsonDeserializationContext jsonDeserializationContext) throws JsonParseException {
         JsonObject jsonObject = jsonElement.getAsJsonObject();
-        JsonElement dataType = jsonObject.get("@type");
-        switch (dataType.getAsString()) {
+        JsonElement dataTypeElement = jsonObject.get("@type");
+        if (dataTypeElement == null) {
+            throw new IllegalArgumentException("Received a service state without a @type property: " + jsonObject);
+        }
+
+        String dataType = dataTypeElement.getAsString();
+        switch (dataType) {
             case "DeviceServiceData" -> {
```

Comment:
Just to be sure, this element should start with uppercase letter, while the other elements should not?

Reply:
This is indeed correct. Some `@type` properties start with upper case and others with lower case. To confirm that `DeviceServiceData` is correct, you can check the `@type` properties in the logs provided [here](https://github.com/openhab/openhab-addons/issues/16590#issuecomment-2052558773).

I just double checked a few log files and also confirmed that `scenarioTriggered`, `userDefinedState` and `message` start with lower case.